pkg/controller/kuberneteseventsource/reconcile.go

```diff
@@ -114,18 +114,17 @@ func (r *reconciler) reconcile(ctx context.Context, source *sourcesv1alpha1.Kube
 		}
 	}
 
-	if cs.Status.IsReady() {
-		source.Status.MarkReady()
-	} else {
-		reason := ""
-		message := ""
-		if ready := cs.Status.GetCondition(sourcesv1alpha1.ContainerConditionReady); ready != nil {
-			reason = ready.Reason
-			message = ready.Message
+	// Update ContainerSource spec if it's changed
+	expected := resources.MakeContainerSource(source, r.receiveAdapterImage)
```

Comment:
nice. In time I would love to bubble this up to the sdk as the children reconciliation/update.